src/N3Quoter.php

```diff
@@ -40,7 +40,49 @@ public function escapeIRI( $iri ) {
 	}
 
 	public function escapeLiteral( $s ) {
-		$escaped = addcslashes( $s, "\x0..\x1F\"\\" );
+		// String escapes. Note that the N3 spec is more restrictive than the Turtle and TR
+		// specifications, see <https://www.w3.org/TeamSubmission/n3/#escaping>
+		// and <https://www.w3.org/TR/turtle/#string>
+		// and <https://www.w3.org/TR/n-triples/#grammar-production-literal>.
+		// Allowed escapes according to the N3 spec are: \r, \n, \t, \', \", \\;
+		// The single quote however does not require escaping when used in double quotes.
+		$escaped = addcslashes( $s, "\r\n\t\"\\" );
+
+		// Handle other control characters using numeric unicode escape sequences
+		$escaped = strtr( $escaped, array(
+			"\x00" => '\u0000',
+			"\x01" => '\u0001',
+			"\x02" => '\u0002',
+			"\x03" => '\u0003',
+			"\x04" => '\u0004',
+			"\x05" => '\u0005',
+			"\x06" => '\u0006',
+			"\x07" => '\u0007',
+			"\x08" => '\u0008',
+			"\x09" => '\u0009',
+			"\x0A" => '\u000A',
```

Comment:
I wonder if we shouldn't use one function here instead of two. Escaping is called tens of millions of times  when generating dump.